Samples/Insights/cpp-win32/InsightsSample.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "InsightsSample.h"
+
+/*
+* Boolean Event
+* 
+* "data": {
+        "value": 1,   <-- PartC
+        "COMMON_WINDOWSAPPSDK_PARAMS": {      <-- PartB
+            "SDKVersion": "0.9",
+            "SDKExperimentationLevel": "",
+            "IsDebugging": false,
+            "AppSessionGuid": "00000D80-0001-0002-B425-C8C4E284D701"
+        }
+    }
+
+* TextPayloadEvent
+* 
+* "data": {
+        "value": "This text is the event payload",
+        "COMMON_WINDOWSAPPSDK_PARAMS": {
+            "SDKVersion": "0.9",
+            "SDKExperimentationLevel": "",
+            "IsDebugging": false,
+            "AppSessionGuid": "00000D80-0001-0002-B425-C8C4E284D701"
+        }
+
+* EventWithUserDefinedType 
+* 
+*  "data": {
+        "value": "This is the user-defined type payload",
+        "COMMON_WINDOWSAPPSDK_PARAMS": {
+            "SDKVersion": "0.9",
+            "SDKExperimentationLevel": "",
+            "IsDebugging": false,
+            "AppSessionGuid": "00000D80-0001-0002-B425-C8C4E284D701"
+        }
+    }
+*/
+
+int main()
+{
+    InsightsSample::BooleanTelemetryEvent(true);
+    InsightsSample::TextPayloadEvent(L"This text is the event payload");
+    InsightsSample::EventWithUserDefinedType(L"This is the user-defined type payload");

```

Comment:
>EventWithUserDefinedType(L"This is the user-defined type payload"); [](http://example.com/codeflow?start=20&length=67)

Would you use a user defined type here, please? It'll help people that'll use this sample as a reference.

Reply:
PCWSTR is technically a user-defined type.